File: cakk-api/src/main/java/com/cakk/api/annotation/DistributedLock.java
package com.cakk.api.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import java.util.concurrent.TimeUnit;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface DistributedLock {


	TimeUnit timeUnit() default TimeUnit.MILLISECONDS;

	long waitTime() default 5000L;

	long leaseTime() default 3000L;
}

File: cakk-core/src/main/kotlin/com/cakk/core/annotation/DistributedLock.kt
package com.cakk.core.annotation

import java.util.concurrent.TimeUnit

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class DistributedLock(

    val key: String = "",
    val timeUnit: TimeUnit = TimeUnit.MILLISECONDS,
    val waitTime: Long = 5000L,
    val leaseTime: Long = 3000L
)
